_posts/2023-11-07-sssd-authentication-method-policy.md

```diff
@@ -0,0 +1,48 @@
+---
+layout:     post
+title:      "SSSD: authentication method policy"
+date:       2023-11-07 12:00:00 +0100
+categories: idm
+---
+
+# Introduction
+
+SSSD has recently released version [2.9.2](https://sssd.io/release-notes/sssd-2.9.2.html), which adds a new option to control the authentication method policy. The need for this new option comes from adding new authentication methods such as passkey[1][2]. It is possible that more authentication methods will emerge in the future, so it seems necessary to add a way to control the policy governing them.
+
+# Existing policy before v2.9.2
+
+When several authentication mechanisms were available SSSD defaulted to try methods that would return a Kerberos ticket. If the server was unavailable, then SSSD would try an offline authentication. For this purpose, it would match the preferred online authentication method, i.e. if the online authentication would be performed via password, the offline authentication would be the same.
+
+Note: Take into account that the offline password authentication workflow requires to set `cache_credentials` to `true` to enable the user cache credential caching. This is potentially security risky, as a malicious actor with access to the cache file (normally requires privileged access) can try to break the password using brute force attack.
+
+# New policy after v2.9.2
+
+The default new policy is the same as before, but the behaviour can be tuned using the `local_auth_policy` domain option. `local_auth_policy` is set to `match` by default, which means that the online and offline methods are matched. Following the example from the previous paragraph, if the passkey method is preferred for online authentication, then the offline one will be handled in the same way.
+
+`local_auth_policy` can also be set to `only`. In this case, even if the server is online, the authentication will be performed offline.

```

Comment:
Maybe it would be worth to list smartcard and passkey authentication here explicitly because iirc currently `local_auth_policy` does not cover password authentication. I have not tested this, but I would expect that with `local_auth_policy = only` local password authentication will not be offered even if there is a password hash stored in the cache.